src/Microsoft.VisualStudio.Threading/JoinableTask.cs

```diff
@@ -332,6 +337,22 @@ internal SynchronizationContext ApplicableJobSyncContext
             }
         }
 
+        /// <summary>
+        /// Gets a weak reference to this object.
+        /// </summary>
+        internal WeakReference<JoinableTask> WeakSelf
```

Comment:
That sounds a cool idea to address this problem.  Thanks a lot to take care of it.